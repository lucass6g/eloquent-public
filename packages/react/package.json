{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
