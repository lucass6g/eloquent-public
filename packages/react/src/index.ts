export {Switch, Case, Default} from './components/conditional-switch'