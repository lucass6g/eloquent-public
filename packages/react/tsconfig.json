{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
