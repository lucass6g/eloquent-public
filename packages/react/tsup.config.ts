import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})