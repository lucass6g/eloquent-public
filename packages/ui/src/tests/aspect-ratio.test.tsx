import { render } from "@testing-library/react";
import { AspectRatio } from "..";

describe("Aspect Ratio", () => {
    describe("should match the snapshot", () => {
        it("default", () => {
            const component = render(<AspectRatio />);

            expect(component.baseElement).toMatchSnapshot()
        })

        it("16 / 9", () => {
            const component = render(<AspectRatio ratio={16 / 9} />);

            expect(component.baseElement).toMatchSnapshot()
        })
    })
}); 