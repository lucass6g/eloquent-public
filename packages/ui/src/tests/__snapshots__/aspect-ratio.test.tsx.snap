// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Aspect Ratio > should match the snapshot > 16 / 9 1`] = `
<body>
  <div>
    <div
      data-radix-aspect-ratio-wrapper=""
      style="position: relative; width: 100%; padding-bottom: 56.25%;"
    >
      <div
        style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
      />
    </div>
  </div>
</body>
`;

exports[`Aspect Ratio > should match the snapshot > default 1`] = `
<body>
  <div>
    <div
      data-radix-aspect-ratio-wrapper=""
      style="position: relative; width: 100%; padding-bottom: 100%;"
    >
      <div
        style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
      />
    </div>
  </div>
</body>
`;
